### PR TITLE
Finish API docs

### PR DESCRIPTION
**Pull request by caspervdw**
I think we might be heading to a next release. It is good to have the API docs finished then.

edit: I renamed some functions to stick with PostGIS namings, those are the namings most GIS experts know.

edit2: the shared_paths functions needs some additional attention (I think it should output 2 arrays or an Nx2 array of geometries), I removed it for now

### CODE REVIEW

File: pygeos/linear.py [hunk 1]
```diff
@@ -1,28 +1,90 @@
...
+    """Returns a point interpolated at given distance on a line.
+
+    Parameters
+    ----------
+    line : Geometry or array_like
+    distance : float or array_like
+        Negative values measure distance from the end of the line. Out-of-range
+        values will be clipped to the line endings.
+    normalize : bool
+        If normalizei is set to True, the distance is a fraction of the total
```
Comment (jorisvandenbossche) [inline pygeos/linear.py]:
```suggestion
        If normalize is set to True, the distance is a fraction of the total
```

File: pygeos/linear.py [hunk 2]
```diff
@@ -1,28 +1,90 @@
...
+    """Returns (multi)linestrings formed by combining the lines in a
+    multilinestrings.
+
+    Parameters
+    ----------
+    line : Geometry or array_like
+
+    Examples
+    --------
+    >>> line = Geometry("LINESTRING(0 2, 0 10)")
```
Comment (jorisvandenbossche) [inline pygeos/linear.py]: this line is not used in the rest of the examples

File: src/ufuncs.c [hunk 3]
```diff
@@ -225,8 +225,26 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
 static PyUFuncGenericFunction Y_Y_funcs[1] = {&Y_Y_func};
 
 /* Define the geom, double -> geom functions (Yd_Y) */
-static void *interpolate_data[1] = {GEOSInterpolate_r};
-static void *interpolate_normalized_data[1] = {GEOSInterpolateNormalized_r};
+static void *GEOSInterpolateProtectEmpty_r(void *context, void *geom, double d) {
+    int n = GEOSGeomGetNumPoints_r(context, geom);
+    if (n < 2) {
+        /* return NULL for empty linestrings to protect against a segfault */
```
Comment (jorisvandenbossche) [inline src/ufuncs.c]: Should this be reported to GEOS, or is it documented that the user needs to check this guarantee?

File: src/ufuncs.c [hunk 4]
```diff
@@ -225,8 +225,26 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
...
+        return NULL;
```
Comment (jorisvandenbossche) [inline src/ufuncs.c]: Other option could also be to return an empty point? (although I think empty points are not very well supported?) 
In general it's always a bit difficult to reason about which of the two is best. But maybe a good rationale is that if GEOS does not return an empty geometry, we shouldn't create one.

**Comment by jorisvandenbossche** [comment]
A general remark on the API docs (but which is not specific to this PR, can also open an issue about it): I personally prefer to document the functions matching the recommended usage.

I *think* we recommend everybody to use `pygeos.union(..)` (or `from pygeos import union`), and not to use `pygeos.set_operations.union(..)` (or `from pygeos.set_operations import union`) ? 
In that case, I think it is clearer for the user to also document that function as `pygeos.union` (the signature at https://pygeos.readthedocs.io/en/latest/set_operations.html#pygeos.set_operations.union gives the long name)

Of course, documenting it as it is done now by module gives a nice structure to the docs. And it makes it easier to do this in sphinx (we can use the `.. automodule::` feature, instead of needing to list the functions per module manually)

File: src/ufuncs.c (hunk 4 above)
Comment (caspervdw) [reply src/ufuncs.c]: I don't know what I would prefer. But I think if GEOS doesn't create an empty geometry, we should do it either.

Comment (caspervdw) [reply src/ufuncs.c]: Check out https://github.com/libgeos/geos/pull/126 , it is already solved upstream. I guess we have to keep the check for backwards compat, but do return an empty point.

**Comment by jorisvandenbossche** [comment]
Updates look good! Maybe add a test case for the empty linestring one? (I know there is a doctest, but I think it is good practice to not rely on those for coverage)
